drivers/clock_control/clock_control_nrf.c

```diff
@@ -222,17 +228,19 @@ static void lfclk_stop(void)
 
 static void hfclk_start(void)
 {
-	if (IS_ENABLED(CONFIG_CLOCK_CONTROL_NRF_SHELL)) {
-		hf_start_tstamp = k_uptime_get();
+	if (IS_ENABLED(CONFIG_CLOCK_CONTROL_NRF_SHELL) &&
+	    IS_ENABLED(CONFIG_NRF_RTC_TIMER)) {
+		hf_start_tstamp = z_nrf_rtc_timer_read();
```

Comment:
This seems to return a 24-bit counter at 32 KiHz, so it wraps at 8 minutes 32 seconds.

Maybe `z_tick_get()` instead, that being the privileged version of `k_uptime_ticks()` (which is going to be faster and more accurate than `k_uptime_get()`).